### ISSUE TITLE
question: logloss residuals computation with raw objective function

### ISSUE THREAD

**Issue by kipukun** [dmlc/xgboost#12117]
hello! i am training a classifier for a binary classification problem using `binary:logitraw` as the objective, with early stopping using `logloss` as the metric. however i am curious how the residuals are computed, since a `predict_proba` call returns raw log odds and not a "probability." does using logloss in this case make sense? thanks!

**Comment by trivialfis** [comment]
It's a "pre-historic" objective when the output base margin wasn't available yet.

**Comment by kipukun** [comment]
it seems to me the best course of action is to use `binary:logistic` and use output margin as necessary. thanks!
